sdk/samples/core/mosquitto_rpc_server_sample.c

```diff
@@ -0,0 +1,293 @@
+  }
+
+  pending_command.correlation_id = AZ_SPAN_EMPTY;
+
+  if (az_result_failed(az_platform_timer_destroy(&timer)))
+  {
+    printf(LOG_APP_ERROR "Failed destroying timer\n");
+    return;
+  }
+}
+
+/**
+ * @brief Start a timer
+ */
+AZ_INLINE az_result start_timer(void* callback_context, int32_t delay_milliseconds)
+{
+  LOG_AND_EXIT_IF_FAILED(az_platform_timer_create(&timer, timer_callback, &callback_context));
+
+  LOG_AND_EXIT_IF_FAILED(az_platform_timer_start(&timer, delay_milliseconds));
+  return AZ_OK;
+}
+
+/**
+ * @brief Stop the timer
+ */
+AZ_INLINE az_result stop_timer()
+{
+  az_result ret = az_platform_timer_destroy(&timer);
+
+  return ret;
+}
+
+/**
+ * @brief Function that does the actual command execution
+ * @note Needs to be modified for your solution
+ */
+az_mqtt5_rpc_status execute_command(az_mqtt5_rpc_server_command_data command_data)
+{
+  // for now, just print details from the command
+  printf(LOG_APP "Executing command to return to: %s\n", az_span_ptr(command_data.response_topic));
+
+  return AZ_MQTT5_RPC_STATUS_OK;
+}
+
+/**
+ * @brief Check if there is a pending command and execute it. On completion, if the command hasn't
+ * timed out, send the result back to the hfsm
+ * @note Result to be sent back to the hfsm needs to be modified for your solution
+ */
+az_result check_for_commands()
+{
+  if (az_span_ptr(pending_command.correlation_id) != NULL)
+  {
+    // copy correlation id to a new span so we can compare it later
+    char copy_buffer[az_span_size(pending_command.correlation_id)];
+    az_span correlation_id_copy
+        = az_span_create(copy_buffer, az_span_size(pending_command.correlation_id));
+    az_span_copy(correlation_id_copy, pending_command.correlation_id);
+
+    az_mqtt5_rpc_status rc = execute_command(pending_command);
+
+    // if command hasn't timed out, send result back
+    if (az_span_is_content_equal(correlation_id_copy, pending_command.correlation_id))
+    {
+      stop_timer();
+      /* Modify the response/error message/status as needed for your solution */
+      az_mqtt5_rpc_server_execution_data return_data
+          = { .correlation_id = pending_command.correlation_id,
+              .response = AZ_SPAN_FROM_STR(
+                  "{\"Succeed\":true,\"ReceivedFrom\":\"mobile-app\",\"processedMs\":5}"),
+              .response_topic = pending_command.response_topic,
+              .status = rc,
+              .error_message = AZ_SPAN_EMPTY };
+      LOG_AND_EXIT_IF_FAILED(az_mqtt5_rpc_server_execution_finish(&rpc_server, &return_data));
+
+      pending_command.correlation_id = AZ_SPAN_EMPTY;
+    }
+  }
+  return AZ_OK;
+}
+
+/**
+ * @brief Callback function for all clients
+ * @note If you add other clients, you can add handling for their events here
+ */
+az_result iot_callback(az_mqtt5_connection* client, az_event event)
+{
+  az_app_log_callback(event.type, AZ_SPAN_FROM_STR("APP/callback"));
+  switch (event.type)
+  {
+    case AZ_MQTT5_EVENT_CONNECT_RSP:
```

Comment:
This event being sent does not always imply we are connected, we would still need to check the `event.data->connack_reason` to make sure we are connected